### ISSUE TITLE
[Feature] Middlewares support for regular expressions

### ISSUE THREAD

**Issue by AbdelrahmanHafez** [Automattic/mongoose#6680]
Now we can set pre/post middlewares to a schema like so:

```js
const catSchema = new Schema({ name: String });

async function middleware () {
  console.log(this._conditions);
  console.log(this._update);
}

catSchema.pre('update', middleware);
catSchema.pre('updateOne', middleware);
catSchema.pre('updateMany', middleware);
catSchema.pre('findOneAndUpdate', middleware);

const Cat = mongoose.model('Cat', catSchema);

Cat.update({ name: 'Fawzy' }, { name: 'Kitty' }).exec();
Cat.updateOne({ name: 'Fawzy' }, { name: 'Kitty' }).exec();
Cat.updateMany({ name: 'Fawzy' }, { name: 'Kitty' }).exec();
Cat.findOneAndUpdate({ name: 'Fawzy' }, { name: 'Kitty' }).exec();
```

It would be great if we can set middlewares using regular expressions like the following as well

```js
catSchema.pre(/update/i, middleware);
```

**Comment by vkarpov15** [comment]
Great idea, will add support for this :+1:

**Comment by vkarpov15** [comment]
Fixed in 5.3 branch
